### PR TITLE
Use global THREE if available

### PR DESCRIPTION
**Pull request by poshaughnessy**
Hi. 

You were spot on - the problem I was having was that I had two instances of THREE:

 - The one that I was pulling in as a standalone script so I could use it in a global way (in order that I could then pull in THREE.ColladaLoader)
 - And the NPM one being required by react-three.

At first I tried switching to just use the NPM one, but I would have had to rewrite the ColladaLoader to be CommonJS compatible. (NB. They're working on this: https://github.com/mrdoob/three.js/issues/4776).
Easier and I think more preferable for now is for me to continue with the recommended browserify-shim way and pull THREE and ColladaLoader in as separate scripts and expose them as globals.

So... Please may I edit this line? All it should do is use the global instance of THREE if available, otherwise require as normal.

This is working for me now, for my react-three-demo.

### CODE REVIEW

**Comment by poshaughnessy** [comment]
Actually - just realised this doesn't quite make sense, so let me double-check on this first!

The idea of browserify-shim is that you should be able to tell Browserify to use a module from the global scope. It looks like doing the require() as normal should be OK. But for some reason it's not - it's pulling THREE in again and bundling it with my Browserify build.

**Comment by poshaughnessy** [comment]
Hmm... It sounds as though you can use browserify-shim for globals, *or* configure it in code via the API (which I want to do for my gulp-driven builds), but not both together, since [they're only available in separate versions](https://github.com/rluba/browserify-global-shim#similar-libraries). I also tried browserify-global-shim and that didn't work either. Finally, I tried switching to SystemJS and using the SFX builder. I got it working if I include everything (including three.js) in my bundle.js, but each build takes ~10 seconds!

I think it would be better for me to pull in the already minified three.js as a separate script include, and either just rely on it being global, or use an [adapter module](https://github.com/systemjs/builder#adapter-modules). But then I think I would need the change in this PR after all, because otherwise react-three will be pulling in the NPM three too.

In other words, I think it would be good for react-three to have this flexibility, so we don't have to rely on people using something like browserify-shim and let them use THREE as a global if they want to. What do you think?

**Comment by Izzimach** [comment]
I dunno, you'll still have that `require('three')` in there so browserify will still pull in the NPM copy of THREE. Your code wlil just not use the NPM version if a global THREE exists. So you'll still have that second copy but won't use it. 

It looks like you might be able to [apply this change](http://stackoverflow.com/questions/26024912/shimming-dependencies-of-dependencies-with-browserify-shim) which will apply the shim to dependencies. You'll have to call the transform in your gulpfile instead of in `package.json` though.

**Comment by poshaughnessy** [comment]
You're spot on again. Just adding {global: true} did it! Thank you very much again. Easy when you know how! :)
